mailscanner/functions.php

```diff
@@ -426,12 +426,16 @@ function html_start($title, $refresh = 0, $cacheable = true, $report = false)
                 } else {
                     echo '    <tr><td colspan="3">' . __('verifyperm03') . ' ' . $incomingdir . ' ' . __('and03') . ' ' . $outgoingdir . '</td></tr>' . "\n";
                 }
-                // else use mailq which is for sendmail and exim
+                // Else use mailq which is for Sendmail and Exim
             } elseif (MAILQ) {
-                $inq = database::mysqli_result(dbquery('SELECT COUNT(*) FROM inq WHERE ' . $_SESSION['global_filter']),
-                    0);
-                $outq = database::mysqli_result(dbquery('SELECT COUNT(*) FROM outq WHERE ' . $_SESSION['global_filter']),
-                    0);
+                // If Exim, use dedicated command, else use database
+               if ($mta === 'exim') {
+                   $inq = exec('sudo /usr/sbin/exim -bpc 2>&1');
+                   $outq = exec('sudo /usr/sbin/exim -bpc -DOUTGOING 2>&1');
```

Comment:
Default for Debian/Ubuntu but not for other system.
Can be configured in the Exim config files but if the user already have a running config, it may be better to have this in conf.php for custom.

As:

define('EXIM_EXEC_INQ', '/usr/sbin/exim -bpc');
define('EXIM_EXEC_OUTQ', '/usr/sbin/exim -bpc -DOUTGOING');